core/src/main/scala/dagr/core/reporting/TopLikeStatusReporter.scala

```diff
@@ -0,0 +1,327 @@
+  def clearScreen: String = "\033[2J"
+  def eraseToEOL: String = "\033[K"
+}
+
+object TopLikeStatusReporter {
+  /** Truncates the line if it is too long and appends "...", otherwise returns the line */
+  private def truncateLine(line: String, maxColumns: Int): String = if (maxColumns < line.length) line.substring(0, line.length-3) + "..." else line
+
+  /** Splits the string by newlines, and truncates each line if necessary.  Enforces a maximum number of lines. */
+  private def wrap(output: String, maxLines: Int, maxColumns: Int): String = {
+    var lines = output.split("\n")
+    lines = if (maxLines <= lines.length) {
+      val numLinesLeft = lines.length - (maxLines-1)
+      lines.slice(0, maxLines-1) ++ List(s"... with $numLinesLeft more lines not shown ...")
+    }
+    else {
+      lines
+    }
+    lines.map(truncateLine(_, maxColumns)).mkString("\n")
+  }
+
+  /** The various columns to report. */
+  object Column extends Enumeration {
+    type Column = Value
+    val Id, Name, Status, Cores, Memory,
+    SubmissionDate, StartDate, EndDate,
+    ExecutionTime, TotalTime, AttemptIndex
+    = Value
+
+    // See: http://stackoverflow.com/questions/2559759/how-do-i-convert-camelcase-into-human-readable-names-in-java
+    def format(column: Column): String = {
+      column.toString.replaceAll(
+        String.format("%s|%s|%s",
+          "(?<=[A-Z])(?=[A-Z][a-z])",
+          "(?<=[^A-Z])(?=[A-Z])",
+          "(?<=[A-Za-z])(?=[^A-Za-z])"
+        ),
+        " "
+      )
+    }
+  }
+
+  /** Creates a new [[TopLikeStatusReporter]] specific to the given executor.  Will add itself to the list of loggers to
+    * be notified by a change in status in [[TaskInfoLike]]. */
+  def apply(executor: Executor): TopLikeStatusReporter = {
+    val loggerOutputStream: ByteArrayOutputStream = {
+      val outStream = new ByteArrayOutputStream()
+      val printStream = new PrintStream(outStream)
+      Logger.out = printStream
+      outStream
+    }
+
+    val logger = executor match {
+      case taskManager: TaskManager      =>
+          new dagr.core.execsystem.TopLikeStatusReporter(
+            taskManager = taskManager,
+            loggerOut   = Some(loggerOutputStream),
+            print       = s => System.out.print(s)
+          )
+      case graphExecutor: GraphExecutor[_] =>
+        new dagr.core.execsystem2.TopLikeStatusReporter(
+          systemResources = graphExecutor.resources.getOrElse(throw new IllegalArgumentException("No resource set defined")),
+          loggerOut       = Some(loggerOutputStream),
+          print           = s => System.out.print(s)
+        )
+      case _ => throw new IllegalArgumentException(s"Unknown executor: '${executor.getClass.getSimpleName}'")
+    }
+
+    // Register!
+    yieldAndThen(logger)(executor.withReporter(logger))
+  }
+}
+
+
+/** A top-like status reporter that prints execution information to the terminal. The
```

Comment:
Note, in a latter commit, I added a lot more tests, including one for the interactive top-like reporter.